android/src/app/organicmaps/widget/placepage/PlacePageController.java

```diff
@@ -392,6 +392,10 @@ public void onPlacePageButtonClick(PlacePageButtons.ButtonType item)
 
   private void onBookmarkBtnClicked()
   {
+    // mMapObject is set to null when the place page closes
+    // We don't want users to interact with the buttons when the PP is closing
+    if (mMapObject == null)
```

Comment:
@arnaudvergnet wouldn't it be a better and more robust solution to disable all PlacePage's buttons interactivity when it's animated?

Reply:
We don't keep a reference to the buttons so its easier to update the click callbacks. We could do the check in `onPlacePageButtonClick` but some callbacks it calls use `mMapObject` so we need to check for null anyway.

Reply:
@arnaudvergnet is it possible to temporarily all input events ("disable?") for parent view and all its subviews, including buttons?

Reply:
I'm asking because there are so many onClick callbacks on the Place Page. Who knows when we get a similar crash in another place? Isn't it better to disable interactivity for the whole fragment when it's animating?

Reply:
Why does this race condition exist? Do we understand what happens and when?

Reply:
Please read the related issue.
When animation starts, mapobject is set to null. But user can tap during the animation. 

Another possible solution btw could be to reset the mapobject only at the end of the animation when PP is hidden. And set it in the beginning of the animation when PP is shown.

Reply:
Let me explain how it works.

This is the current structure: The PlacePageController controls 2 fragments, the place page buttons (the bottom row of buttons) and the place page view (the actual place page content you can swipe).

The map object is stored in the viewmodel so the fragments can access it and listen to its changes. When the user clicks on the map, we set the map object on the viewmodel. The PlacePageController listens to this change, creates the buttons and the place page view fragment then opens it. When the user swipes down the place page, the PlacePageController will detect that and set the map object to null. This use case works well: if the map object is null, then the place page is closed

But it becomes more complex when the user requests to close the place page without swiping down (eg: clicking on an empty part of the map). The current behavior is to set the map object to null from the main activity (where we handle map interaction), the PlacePageController detects the change and plays the close animation. During the animation the buttons and the place page view are still visible for a short time. This PR fixes the issue of the place page buttons callbacks still referring to the destroyed map object.

After testing a bit more there is another issue. If the user is fast enough, they can grab the place page view before it can be fully hidden. In that case the user can keep the place page open but without a valid map object reference in the viewmodel. They can then interact with the place page view and the buttons. The buttons will not work because we disabled the callbacks if the viewmodel's map object is null. But the place page view will still work as we ignore viewmodel changes in this fragment when map object becomes null (because that view is supposed to be destroyed when map object is null).

I'll try to find a way to disable user interactions on the place page view when the close animation is playing.

Reply:
Resetting the map object only when the PP is fully closed is another solution, but one which would require another way to control opening/closing the PP.

Reply:
Aha, thanks for the explanation! Looks like it's a limitation of the currently used approach (use null as a signal to action)…